### PR TITLE
imagetools resolver: discard containerd logger output

### PR DESCRIPTION
**Pull request by crazy-max**
Discard containerd logger in imagetools resolver to avoid printing unnecessary info during image reference resolution:

```console
$ docker buildx imagetools inspect alpinefoo
INFO[0001] trying next host                              error="pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed" host=registry-1.docker.io
error: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
```

now:

```console
$ docker buildx imagetools inspect alpinefoo
error: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
```

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: util/imagetools/inspect.go [hunk 1]
```diff
@@ -55,6 +57,12 @@ func (r *Resolver) resolver() remotes.Resolver {
 }
 
 func (r *Resolver) Resolve(ctx context.Context, in string) (string, ocispec.Descriptor, error) {
+	// discard containerd logger to avoid printing unnecessary info during image reference resolution.
+	// https://github.com/containerd/containerd/blob/1a88cf5242445657258e0c744def5017d7cfb492/remotes/docker/resolver.go#L288
+	logger := logrus.New()
+	logger.Out = io.Discard
+	ctx = log.WithLogger(ctx, logrus.NewEntry(logger))
+
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply util/imagetools/inspect.go:65]: Yes I had started to use a dedicated pkg for that in https://github.com/docker/buildx/pull/854/commits/df28fa56a792101cfc4cafad3dec7fd7c461c317 but containerd is not used anywhere else in this code base. I can move the discarder in a specific func if you prefer.

[the remark this answers is not in the document]
Comment (thaJeztah) [reply util/imagetools/inspect.go:65]: If this is the only location, it's probably ok to keep it as-is. It's a pity that the containerd code is tied to `logrus.Entry` (and not some generic interface), but it is what it is 🤷‍♂️

[the remark this answers is not in the document]
Comment (thaJeztah) [reply util/imagetools/inspect.go:65]: Perhaps we should have a look if we can contribute something to containerd to make the default logger a "discard" logger, and require it to be explicitly set up if someone wants to use logging.

Alternatively; have a `log.WithLogDiscarder()` (e.g.) utility

**Review by thaJeztah — APPROVED** [review]
LGTM as a quick 'n dirty fix

**Pull request merged**